src/app/components/cohort-selection/cohort-selection.service.js

```diff
@@ -7,169 +7,204 @@
+                var children = elm.children();
+                var mainContainer = null;
+                for (var i = 0; i < children.length; i++) {
+                    var child = children[i];
+                    if (angular.element(child).hasClass('main-chart-container')) {
+                        mainContainer = angular.element(child);
+                        break;
+                    }
                 }
-            });
-            return foundChart;
-        };
-
-        /**
-         * @memberof CohortSelectionService
-         * @param path - conceptPath
-         * @param nodes - the array of nodes in ctrl.cs.nodes
-         * @returns {Object} - the found node
-         */
-        service.findNodeByConceptPath = function (path, nodes) {
-            var foundNode = undefined;
-            nodes.forEach(function (node) {
-               if(node.label.conceptPath === path) {
-                   foundNode = node;
-               }
-            });
-            return foundNode;
-        };
-
-        /**
-         * @memberof CohortSelectionService
-         * @param value
-         * @returns {string} - The type of label
-         */
-        service.getLabelType = function (value) {
-            var _type = typeof value;
-            if (_type === 'string') {
-                if (value === 'E' || value === 'MRNA') {
-                    _type = 'highdim';
+                var id = boxId + '-main-chart-container';
+                if (mainContainer !== null) {
+                    mainContainer.attr('id', id);
                 }
-            } else if (_type === 'number') {
-                if ((value % 1) !== 0) {
-                    _type = 'float';
+                return id;
+            };
+
+            /**
+             * @memberof CohortSelectionService
+             * @param {String} path - conceptPath
+             * @param {Array} charts - the array of charts to be searched
+             * @returns {*} - The found chart in CohortSelectionCtrl.cs.charts,
+             *      with matching name chartName, if not found, return null
+             */
+            service.findChartByConceptPath = function (path, charts) {
+                var foundChart = null;
+                charts.forEach(function (_chart) {
+                    if (_chart.tsLabel.conceptPath === path) {
+                        foundChart = _chart;
+                    }
+                    else if (_chart.tsLabel.type === 'combination'
+                        && _chart.tsLabel.name === path) {
+                        foundChart = _chart;
+                    }
+                });
+                return foundChart;
+            };
+
+            /**
+             * @memberof CohortSelectionService
+             * @param path - conceptPath
+             * @param nodes - the array of nodes in ctrl.cs.nodes
+             * @returns {Object} - the found node
+             */
+            service.findNodeByConceptPath = function (path, nodes) {
```

Comment:
there is the extra utility in this function to finding a combination chart by its name, as shown in the iteration. So it is not a standard _.find search.